<?xml version="1.0"?>
<launch>
  <arg name="local_planner" default="mpc"/>
  
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
		<rosparam file="$(find sdv_test)/params/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
		<rosparam file="$(find sdv_test)/params/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find sdv_test)/params/move_base_config/local_costmap_params.yaml" command="load" />
		<rosparam file="$(find sdv_test)/params/move_base_config/global_costmap_params.yaml" command="load" />
		<rosparam file="$(find sdv_test)/params/move_base_config/move_base_params.yaml" command="load" />
		<rosparam file="$(find sdv_test)/params/move_base_config/global_planner.yaml" command="load" />	
		<!-- <param name="base_global_planner" type="string" value="carrot_planner/CarrotPlanner" /> -->
		<param name="base_global_planner" type="string" value="navfn/NavfnROS" />

		<!-- <param name="base_local_planner" value="forx_local_planner/MpcLocalPlanner"  if="$(eval local_planner == 'mpc')"/> -->

		 <param name="base_local_planner" value="local_planner/LocalPlanner" />




		
		<!-- <param name="base_global_planner" type="string" value="global_planner/GlobalPlanner" /> -->

		<!-- <param name="SBPLLatticePlanner/primitive_filename" value="$(find sdv_test)/params/move_base_config/unicycle.mprim" />
		<rosparam file="$(find sdv_test)/params/move_base_config/sbpl_global_params.yaml" command="load" />
		<param name="base_global_planner" type="string" value="SBPLLatticePlanner" /> -->

		<!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"  if="$(eval local_planner == 'teb')"/>
		<rosparam file="$(find sdv_test)/params/move_base_config/teb_carlike_params.yaml" command="load" if="$(eval local_planner == 'teb')" />
		
		<param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"  if="$(eval local_planner == 'teb_')"/>
		<rosparam file="$(find sdv_test)/params/move_base_config/teb_local_planner_params.yaml" command="load" if="$(eval local_planner == 'teb_')" />

		<param name="base_local_planner" value="mpc_local_planner/MpcLocalPlannerROS"  if="$(eval local_planner == 'mpc')"/>
		<rosparam file="$(find sdv_test)/params/move_base_config/mpc_local_planner_simple_car.yaml" command="load" if="$(eval local_planner == 'mpc')" />

		<rosparam file="$(find sdv_test)/params/move_base_config/pure_pursuit_controller.yaml" command="load" if="$(eval local_planner == 'pure_pursuit')"/>
		<param name="base_local_planner" value="local_planner/PurePursuitController" if="$(eval local_planner == 'pure_pursuit')"/>


		<remap from="cmd_vel" to="/tricycle_controller/cmd_vel3" /> -->
	</node>
</launch>
